packages/kit/src/runtime/client/client.js

```diff
@@ -312,16 +312,17 @@ async function _invalidate() {
 
 	if (navigation_result) {
```

Comment:
you could change the previous line to `if (nav_token !== token  || !navigation_result) return;` to simplify things